### DISCUSSION TITLE
Getting ActionCable to work with Turbo

### DISCUSSION THREAD

**Discussion by chmich** [ElMassimo/vite_ruby#364] [Show and tell]
**Hi**

`vite:install` generates these two lines in `application.js`:

```js
//import * as Turbo from '@hotwired/turbo'
//Turbo.start()
```

For using Turbo on its own that works.

When using `action_cable`, you need to avoid tree-shaking, which you can do by changing the import to:

```js
import '@hotwired/turbo-rails'
```

Related Issue in `turbo-rails`:

- https://github.com/hotwired/turbo-rails/issues/48

**Comment by ElMassimo** [comment]
Hi Christian!

Thanks, this info might be useful to someone setting up ActionCable, so I'll convert it to a _Discussion_.

I think I'd prefer dropping these "examples" from the installation script, as they will likely change and need to be updated over the years. In addition, not every Vite Rails user is using these libraries, and I don't want beginners to assume they are necessary (nor recommended).
